smartlearner/tasks.py

```diff
@@ -23,6 +25,15 @@ def init(self, status):
     def finished(self, status):
         print("Training done in {:.03f} sec.".format(time() - self.start_time))
 
+    def save(self, path):
+        state = {"version": 1,
+                 "start_time": self.start_time}
+        utils.save_dict_to_json_file(pjoin(path, "print_training_duration.json"), state)
```

Comment:
Actually we should used JSON everywhere, especially since I've added a `NumpyEncoder` and `json_numpy_obj_hook` in `smartlearner/utils.py`.

One reason sometime I don't is weights matrices filled up the JSON and is not human readable anymore. So I tend to save the hyperparams in JSON and parameters like the weights of the model and ADAGRAD's parameters to .npy.

What do you think?